docs/details/other-components/xml.rst

```diff
@@ -0,0 +1,301 @@
+An example of a ``my_button`` component looks like this:
+
+.. code-block:: xml
+
+    <component>
+        <consts>
+            <px name="size" value="100"/>
+            <color name="orange" value="0xffa020"/>
+        </consts>
+
+        <params>
+            <string name="btn_text"/>
+        </params>
+
+        <styles>
+            <style name="blue" bg_color="0x0000ff" radius="2"/>
+            <style name="red" bg_color="0xff0000"/>
+        </styles>
+
+        <view extends="lv_button" width="#size" styles="blue red:pressed">
+            <my_h3 text="$btn_text" align="center" color="#orange" style_text_color:checked="0x00ff00"/>
+        </view>
+    </component>
+
+- ``<component>``: The root element.
+- ``<consts>``: Constants with ``px`` or ``color`` types.
+Constants can later be referenced as ``#name``.
+- ``<params>``: Parameters with ``int``, ``px``, ``string``, ``color``, or ``style`` types.
+Parameters can later be referenced as ``$name``.
+- ``<styles>``: ``<style>`` properties can be defined with names and properties.
+- ``<view>``: Describes how the component looks. Can reference constants, parameters, and styles.
+
+Naming conventions:
+
+- A standard XML syntax is used.
+- Lowercase letters with ``_`` separation are used for attribute names.
+- The usual variable name rules apply for attribute and tag names: only letters, numbers, `'_'` and can't start with a number.
+- LVGL API is followed as much as possible, e.g., ``align="center"``, ``bg_color="0xff0000"``.
+- ``params`` can be referenced with ``$``
+- ``consts`` can be referenced with ``#``
+- ``styles`` can be attached to states and/or parts like ``styles="red blue:pressed green:focused:scrollbar"``
+- Local styles can be used like  ``<lv_label style_text_color="0xff0000" style_text_color:checked="0x00ff00" ``
+Usage
+~~~~~
+
+Once a component is created (e.g., ``my_button``), it can be registered by calling either:
+
+- ``lv_xml_component_register_from_file("A:lvgl/examples/others/xml/my_button.xml");``
+- ``lv_xml_component_register_from_data("my_button", xml_data_of_my_button);``
+
+These registration functions process the XML data and save some relevant data internally.
+This is required to make LVGL recognize the components by name.
+When loaded from a file, the file name is used as the component name.
+
+After this, a new instance of any of the registered components can be created with:
+``lv_obj_t * obj = lv_xml_create(lv_screen_active(), "my_button", NULL);``
+
+The created widget is a normal LVGL widget that can be used like any other manually created widget.
+
+The last parameter can be ``NULL`` or an attribute list, like this:
+
+.. code-block:: c
+
+    /* Can be local */
+    char * my_button_attrs[] = {
+        "x", "10",
+        "y", "-10",
+        "align", "bottom_left",
+        "btn_text", "New button",
+        NULL, NULL,
+    };
+
+    lv_obj_t * btn1 = lv_xml_create(lv_screen_active(), "my_button", my_button_attrs);
```

Comment:
Feels like the lv_xml_create is bounded to xml, but the functionality can be in a layer that can be reused by other bindings, just like xml.

Reply:
I see what you mean. Probably it can be really unified by obj_prop on this level. Let's reconsider it in a new round.